babel/src/test/java/org/apache/calcite/test/BabelParserTest.java

```diff
@@ -328,6 +328,54 @@ private void checkParseInfixCast(String sqlType) {
     sql(sql).ok(expected);
   }
 
+  @Test public void testTableAttributeJournalDefault() {
+    final String sql = "create table foo, journal (bar integer)";
+    final String expected = "CREATE TABLE `FOO`, JOURNAL (`BAR` INTEGER)";
+    sql(sql).ok(expected);
+  }
+
+  @Test public void testTableAttributeJournalBefore() {
+    final String sql = "create table foo, before journal (bar integer)";
+    final String expected = "CREATE TABLE `FOO`, BEFORE JOURNAL (`BAR` INTEGER)";
+    sql(sql).ok(expected);
+  }
+
+  @Test public void testTableAttributeJournalAfter() {
+    final String sql = "create table foo, after journal (bar integer)";
+    final String expected = "CREATE TABLE `FOO`, AFTER JOURNAL (`BAR` INTEGER)";
+    sql(sql).ok(expected);
+  }
+
+  @Test public void testTableAttributeJournalNoModifier() {
```

Comment:
testTableAttributeJournalNoModifier can be change to testTableAttributeJournal_noModifier_before

Reply:
My slight issue with using this naming convention is that it's different from the style of the other test names. Not that I dislike this proposed convention, I think it's actually quite nice, but this style difference is something I think we should consider.

Reply:
I do like that it specifies it more and I see that style is in this [style guide](https://google.github.io/styleguide/javaguide.html#s5.2.3-method-names), but I'm not sure if that would go against the style used in this project. Maybe @dasch-1 could clarify that

Reply:
IMO, where there's an existing style in Calcite, we should try to match it. If there's no existing pattern to follow, however, I think going with whatever is recommended in the style guide makes sense.

Reply:
I am not recommending a particular naming style, instead, I use underscore with the intention to show that idea of combining states together in the test name. As long as you get the gist, the actual naming style could varies

Reply:
I think the underscores in the name is part of the style. And we should try to match the existing style. The alternative of trying to rewrite all of the existing test names in the new style seems not worth it to me :)

Reply:
Yeah, the idea is not rewrite the existing test with the underscore style. It is from a previous conversation that I suggested to include both of JournalModifier and JournalType in the test name. Some test names in the current version is missing one of those two and that is what I think can be change. @dasch-1

Reply:
I can change the names so that it has modifier then type but in camel case so it matches the other tests. Something like testTableAttributeJournalNoModifierDefaultType

Reply:
SGTM